src/CodeGen_Xtensa.cpp

```diff
@@ -858,6 +896,46 @@ HALIDE_ALWAYS_INLINE native_vector_i32_x2 load_predicated<native_vector_i32_x2,
     return *((native_vector_i32_x2 *)output);
 }
 
+template <>
+HALIDE_ALWAYS_INLINE native_vector_f32_x2 load_predicated<native_vector_f32_x2, native_vector_i32_x2, native_mask_i16, float, 2 * VECTOR_WIDTH_F32>(const void *base, const native_vector_i32_x2& offset, const native_mask_i16& predicate) {
+    int __attribute__((aligned(XCHAL_VISION_SIMD8))) offsets[2 * VECTOR_WIDTH_F32];
+    aligned_store<native_vector_i32_x2, int32_t, 2 * VECTOR_WIDTH_F32>(offset, &offsets[0], 0);
+    native_vector_u16 vmask = IVP_MOVNX16T(native_vector_u16(1), native_vector_u16(0), predicate);
+    uint8_t __attribute__((aligned(XCHAL_VISION_SIMD8))) mask[2 * VECTOR_WIDTH_F32];
+    aligned_store<native_vector_u16, uint16_t, 2 * VECTOR_WIDTH_F32>(vmask, &mask[0], 0);
+
+    float __attribute__((aligned(XCHAL_VISION_SIMD8))) output[2 * VECTOR_WIDTH_F32];
+    for (int i = 0; i < 2 * VECTOR_WIDTH_F32; i++) {
+        if (mask[i] == 1) {
+            output[i] = ((const float*)base)[offsets[i]];
+        } else {
+            output[i] = 0;
```

Comment:
Does mask=0 for a predicated load mean "set this entry to zero in the output", or "leave this entry unchanged in the output"? In predicated_store, mask=0 definitely means "leave this entry unchanged".

Reply:
Found the same behaviour in other load_predicated variants in the same file

Reply:
I hear you, but I'm not sure if those are right, either; these are meant to implement the `Load` IR node in Halide, and the implementation of these in Codegen_C leaves the entry unchanged for mask=0. I'm not sure if that's what the LLVM implementation does as well, but I rather strongly suspect so. Adding @abadams to comment.

Reply:
They should be set to an undefined value, so zero is safe. That's what the llvm backend does.

Reply:
OK, in that case I'd argue that

`output[i] = mask[i] ? ((const float*)base)[offsets[i]] : 0;` is slightly better style, but probably results in identical codegen